### ISSUE TITLE
[Android] Order returns error

### ISSUE THREAD

**Issue by HarelM** [j3k0/cordova-plugin-purchase#1031]
### system info

Please report your OS, OS version, Compiling System and Version, used Plugin Version and any helpful information that can help us to resolve the problem.

In example:
```

Model: Redmi 4X
Platform: Android
OS version: 7.1.2
Cordova 9.0
Plugin Version: 10.1.1
```

### Expected behavior

When calling the `order` method the store dialog should pop-up

### Observed behavior

Nothing happens, there's an error in the logs

### Steps to reproduce
Call order with latest released to npm version of the plugin.

Should I downgrade the plugin to a previous version? If so, which?

The following is the log file from the device:
```
2020-05-05 14:08:00 |  INFO | Starting IHM Application Initialization
2020-05-05 14:08:00 | DEBUG | Original brightness is: -1.0
2020-05-05 14:08:00 | DEBUG | [Store] state: offline_map -> registered
2020-05-05 14:08:00 | DEBUG | [Store] store.trigger -> triggering action refreshed
2020-05-05 14:08:00 | DEBUG | [Store] refresh -> checking products state (1 products)
2020-05-05 14:08:00 | DEBUG | [Store] refresh -> product id offline_map (offline map)
2020-05-05 14:08:00 | DEBUG | [Store]            in state 'registered'
2020-05-05 14:08:00 | DEBUG | [Store] store.trigger -> triggering action re-refreshed
2020-05-05 14:08:00 | DEBUG | [Store] iabGetPurchases()
2020-05-05 14:08:00 |  INFO | Finished IHM Application Initialization
2020-05-05 14:08:01 | DEBUG | [Store] store.trigger -> triggering action refresh-completed
2020-05-05 14:09:17 | DEBUG | [Store] store.trigger -> triggering action refreshed
2020-05-05 14:09:17 | DEBUG | [Store] refresh -> checking products state (1 products)
2020-05-05 14:09:17 | DEBUG | [Store] refresh -> product id offline_map (offline map)
2020-05-05 14:09:17 | DEBUG | [Store]            in state 'registered'
2020-05-05 14:09:17 | DEBUG | [Store] store.trigger -> triggering action re-refreshed
2020-05-05 14:09:17 | DEBUG | [Store] iabGetPurchases()
2020-05-05 14:09:17 | DEBUG | [Store] iabSetPurchases: []
2020-05-05 14:09:17 | DEBUG | [Store] inappbilling.getPurchases() -> Success
2020-05-05 14:09:17 | DEBUG | [Store]                             -> object
2020-05-05 14:09:17 | DEBUG | [Store]                             -> []
2020-05-05 14:09:17 | DEBUG | [Store] iabUpdatePurchases: []
2020-05-05 14:09:17 | DEBUG | [Store] state: offline_map -> valid
2020-05-05 14:09:17 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "valid",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": true,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": null,
    "transaction": null,
    "valid": true
}
2020-05-05 14:09:17 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "valid",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": true,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": null,
    "transaction": null,
    "valid": true
}
2020-05-05 14:09:17 | DEBUG | [Store] store.trigger -> triggering action refresh-completed
2020-05-05 14:09:27 | DEBUG | [Store] Ordering product for: 6626119
2020-05-05 14:09:27 | DEBUG | [Store] state: offline_map -> requested
2020-05-05 14:09:27 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "requested",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": false,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
2020-05-05 14:09:27 | DEBUG | [Store] state: offline_map -> initiated
2020-05-05 14:09:27 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "initiated",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": false,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
2020-05-05 14:09:27 |  INFO | [Store] plugin -> subscribe error 6777003
2020-05-05 14:09:27 | ERROR | [Store] error handler: Purchase failed: Product not registered: offline_map (6777003)
2020-05-05 14:09:27 | DEBUG | [Store] state: offline_map -> valid
2020-05-05 14:09:27 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "valid",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": true,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
2020-05-05 14:09:49 | DEBUG | Pausing app, stopping user idle service
2020-05-05 14:09:54 | DEBUG | Resume app, setting brightness to original: -1.0
2020-05-05 14:09:55 | DEBUG | Pausing app, stopping user idle service
2020-05-05 14:11:06 | DEBUG | [Store] iabSetPurchases: []
2020-05-05 14:11:06 | DEBUG | [Store] iabUpdatePurchases: []
2020-05-05 14:11:06 | DEBUG | Resume app, setting brightness to original: -1.0
2020-05-05 14:11:51 | DEBUG | [Store] Ordering product for: 6626119
2020-05-05 14:11:51 | DEBUG | [Store] state: offline_map -> requested
2020-05-05 14:11:51 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "requested",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": false,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
2020-05-05 14:11:51 | DEBUG | [Store] state: offline_map -> initiated
2020-05-05 14:11:51 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "initiated",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": false,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
2020-05-05 14:11:51 |  INFO | [Store] plugin -> subscribe error 6777003
2020-05-05 14:11:51 | ERROR | [Store] error handler: Purchase failed: Product not registered: offline_map (6777003)
2020-05-05 14:11:51 | DEBUG | [Store] state: offline_map -> valid
2020-05-05 14:11:51 | DEBUG | [Store] Updated: offline_map
{
    "id": "offline_map",
    "alias": "offline map",
    "type": "paid subscription",
    "group": "default",
    "state": "valid",
    "title": null,
    "description": null,
    "priceMicros": null,
    "price": null,
    "currency": null,
    "countryCode": null,
    "loaded": true,
    "canPurchase": true,
    "owned": false,
    "introPrice": null,
    "introPriceMicros": null,
    "introPricePeriod": null,
    "introPriceNumberOfPeriods": null,
    "introPricePeriodUnit": null,
    "introPriceSubscriptionPeriod": null,
    "introPricePaymentMode": null,
    "ineligibleForIntroPrice": null,
    "discounts": [],
    "downloading": false,
    "downloaded": false,
    "additionalData": {
        "applicationUsername": 6626119,
        "accountId": "fa0b94e0ed25349743e7450e9ace3087"
    },
    "transaction": null,
    "valid": true
}
```

**Comment by HarelM** [comment]
Probably related/duplicate of #706 #744 but this was working as expected when using a previous version (10.0.1).
I have made some changes to add `applicationUsarename` so now it is provided in both setting it on the `store` object and adding it to the additional data in the `order` method.
Code can be seen here:
https://github.com/IsraelHikingMap/Site/blob/master/IsraelHiking.Web/sources/application/services/purchase.service.ts#L24
Main changes can be seen here:
https://github.com/IsraelHikingMap/Site/commit/ea033aab4be5155655fa58282cde861d95238a50#diff-a45888d8aabe3c6310a99eb094cb0777

**Comment by HarelM** [comment]
I've done a lot of trial and error. The following is what I see right now:
1. I've reverted to a previous version, it didn't help. The problem is not with the latest version.
2. I've reverted the refresh and set `applicationUsername` after user login and the problem was solved.
3. When calling refresh before setting `applicationUsername` and then calling refresh after setting the `applicationUsername` the problem seems to happen, I think...

So 3 is probably the root cause of this issue.
It only happens on Android.
